### PR TITLE
feat: Docker setup, and component logic improvements

### PR DESCRIPTION
**Pull request by MuhammadAli-Paysys**
# SPDX-License-Identifier: Apache-2.0

This pull request introduces several improvements across the backend and test suites, focusing on increasing code coverage, improving error handling, and enhancing test robustness. It also adds the frontend service to the Docker Compose configuration for a more integrated development environment.

**Testing and Coverage Improvements:**

- Increased the global test coverage threshold to 95% in `jest.config.ts` to enforce higher code quality.
- Added comprehensive new test cases to the `NotificationService` and `ConfigWorkflowService` to check for edge cases, error handling, and fallback behaviors. This includes tests for unknown errors, missing configuration, fallback values, and various action/permission scenarios. [[1]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R166-R190) [[2]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R274-R381) [[3]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R413-R432) [[4]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R501-R517) [[5]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R571-R590) [[6]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R763-R782) [[7]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R906-R982) [[8]](diffhunk://#diff-9ca47f8b52683caadfc30f1dbb0a0c01f9589af4ec48487d020d9acc16648caaR186-R195) [[9]](diffhunk://#diff-9ca47f8b52683caadfc30f1dbb0a0c01f9589af4ec48487d020d9acc16648caaR238-R324) [[10]](diffhunk://#diff-d12da395d612a2af2c722ea5458df42a8a9dc92974611b088d0c478f3209f1b9R503-R517)

**Error Handling and Code Instrumentation:**

- Added `c8 ignore` comments to constructors in several service classes (e.g., `AuthService`, `JobService`, `NotificationService`, `NotifyService`, `SchedulerService`, `SimulationService`, `TazamaDataModelService`) to exclude boilerplate dependency injection code from coverage metrics, making coverage reports more meaningful. [[1]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398R14) [[2]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899R45) [[3]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899R55) [[4]](diffhunk://#diff-031c93bd71a05e5bc9089506d7c6d6d3b049791b6f80a94b3cab783fa8719056R66-R71) [[5]](diffhunk://#diff-64b90b61c762f9cde05942ffed78e30a37cab268e40e36678ea75049eea92397R19-R24) [[6]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eR28-R35) [[7]](diffhunk://#diff-5626bbd722e780683b056f62de2d91fc05ecfdc3d43f00ebcfa9d4fb60b911e6R26-R28) [[8]](diffhunk://#diff-586d81c3a2f8fe6e5c4c10a9feda9e884b8a621c49c0d5159851d9a5652b0a82R15-R17)

**Bug Fixes and Minor Logic Updates:**

- Fixed a logic bug in `SimulationService` where the check for "has only text and attributes" was incorrectly allowing keys that did not start with `@`.

**Development Environment:**

- Added the `connection-studio-frontend` service to the `docker-compose-tcs.yml` file, enabling the frontend to be built and run alongside the backend for a smoother local development experience.

---

**Detailed List of Most Important Changes:**

**Testing and Coverage Enhancements**
- Increased Jest global coverage thresholds to 95% for branches, functions, lines, and statements in `backend/jest.config.ts`.
- Added extensive tests for `NotificationService`, including edge cases for missing configuration, unknown errors, and fallback behaviors. [[1]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R166-R190) [[2]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R274-R381) [[3]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R413-R432) [[4]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R501-R517) [[5]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R571-R590) [[6]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R763-R782) [[7]](diffhunk://#diff-e186a203ab951dcf20a47a579751c44499fd3b319ea339aa601bd2519f1fd376R906-R982)
- Expanded tests for `ConfigWorkflowService` to cover additional transitions, permission checks, and unknown actions. [[1]](diffhunk://#diff-9ca47f8b52683caadfc30f1dbb0a0c01f9589af4ec48487d020d9acc16648caaR186-R195) [[2]](diffhunk://#diff-9ca47f8b52683caadfc30f1dbb0a0c01f9589af4ec48487d020d9acc16648caaR238-R324)
- Added a test to `DryRunService` to ensure default comma delimiter is used for CSV files when not specified.

**Error Handling and Code Instrumentation**
- Added `c8 ignore` comments to constructors in multiple backend services to exclude dependency injection boilerplate from coverage. [[1]](diffhunk://#diff-34c4ae148a4bf99d6ec2000c464097cd50c195fbb68082b4e13dab69ea96b398R14) [[2]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899R45) [[3]](diffhunk://#diff-54234b797575b2b714ddbf97ebd00222d66303f2a5bee14cb6ccd14ecec7f899R55) [[4]](diffhunk://#diff-031c93bd71a05e5bc9089506d7c6d6d3b049791b6f80a94b3cab783fa8719056R66-R71) [[5]](diffhunk://#diff-64b90b61c762f9cde05942ffed78e30a37cab268e40e36678ea75049eea92397R19-R24) [[6]](diffhunk://#diff-4d6aa05987e7164e55cc74f1a6988fdd588e3a6a7a678cf6b544e7bfd267fa7eR28-R35) [[7]](diffhunk://#diff-5626bbd722e780683b056f62de2d91fc05ecfdc3d43f00ebcfa9d4fb60b911e6R26-R28) [[8]](diffhunk://#diff-586d81c3a2f8fe6e5c4c10a9feda9e884b8a621c49c0d5159851d9a5652b0a82R15-R17)

**Bug Fixes**
- Corrected a logic error in `SimulationService` for checking object keys, ensuring attributes are properly detected.

**Development Environment**
- Added `connection-studio-frontend` to Docker Compose, facilitating integrated local development of frontend and backend.

---

**Note**
- Test case coverage and lint issues fill will be fixed in the upcoming PR's.
- Test cases are passing

image

image

---

## How was it tested?
- [X] Locally
- [X] Development Environment
- [ ] Not needed, changes very basic
- [X] Husky successfully run
- [X] Unit tests passing and Documentation done

## Summary by CodeRabbit

* **New Features**
  * Frontend available as a Docker Compose service.
  * Job approval dialog with comment and loading-state UX.

* **Improvements**
  * Action buttons disable during in-flight approve/reject.
  * Schedule lists now show only approved/exported schedules.
  * Stronger validation for SFTP paths, HTTP URLs, headers and endpoint fields.
  * Cleaner error messages and standardized HTTP/SFTP error handling.

* **Bug Fixes**
  * XML normalization for text-only nodes corrected.
  * CSV parsing defaults to comma delimiter when unspecified.

* **Tests**
  * Extensive test coverage added across many modules.

* **Chores**
  * Added repository tooling: .gitignore, npmrc, and Git hooks.

* **Removed**
  * Dashboard stats cards and related UI/tests removed.

### CODE REVIEW

File: frontend/src/features/cron/hooks/useCronJobList.ts [hunk 1]
```diff
@@ -70,18 +82,19 @@ export const useCronJobList = (): {
...
       limit,
       totalRecords: total,
-      setPage: (page: number) => { setOffset(page - PAGE_OFFSET_ADJUSTMENT); },
-    }), [offset, limit, total, setOffset])  
+      setPage: (page: number) => {
+        setOffset(page - PAGE_OFFSET_ADJUSTMENT);
+      },
+    }),
+    [offset, limit, total, setOffset],
+  );
```
[the remark this answers is not in the document]
Comment (MuhammadAli-Paysys) [reply frontend/src/features/cron/hooks/useCronJobList.ts:97]: Page is used as offset in CustomTable component

File: frontend/src/features/dashboard/components/DashboardBoxes.tsx [hunk 2]
```diff
@@ -36,7 +36,7 @@ const BoxCard: React.FC<{
         gap: 2,
         textAlign: 'center',
         position: 'relative',
-        backgroundColor: (theme: any) => theme.palette.background.paper,
+        backgroundColor: color,
```
[the remark this answers is not in the document]
Comment (MuhammadAli-Paysys) [reply frontend/src/features/dashboard/components/DashboardBoxes.tsx]: This changes this color theme of my design so no need of this.

**Review by MAdeel95 — APPROVED** [review]
